tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,163 @@
+        if rbd.image_resize(pool, image, size, **encryption_passphrase):
+            log.error(
+                f"Image resize to compensate overhead due to {encryption_type} header failed for {image_spec}"
+            )
+
+        encryption_config = [{encryption_type: passphrase}]
+        out, err = rbd.device_map(image_spec, "nbd", encryption_config)
+        log.info(f"Out, Err : {out}: {err}")
+        if err:
+            log.error(
+                f"RBD device map failed for {image_spec} and encryption_config: {encryption_config}"
+            )
+
+        io_config = {
+            "file_size": "100M",
+            "image_spec": [f"{image_spec}"],
+            "operations": {
+                "fs": "ext4",
+                "io": True,
+                "mount": True,
+                "nounmap": True,
+            },
+            "runtime": 30,
+            "device_names": [out],
+        }
+        kw["config"].update(io_config)
+        kw.update({"rbd_obj": rbd, "do_not_create_image": True})
+        krbd_io_handler(**kw)
+
+        log.info(f"Creating snap and clone for {image_spec}")
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Snapshot with name {snap} creation failed for {image_spec}")
+
+        if rbd.protect_snapshot(snap_spec):
+            log.error(f"Snapshot protect failed for {image_spec}")
+
+        krbd_io_handler(**kw)
+
+        if rbd.create_clone(snap_spec, pool, clone):
+            log.error(f"Clone creation failed for {clone_spec}")
+
+        log.info(
+            f"Applying encryption of type {encryption_type} on the clone {clone_spec}"
+        )
+        clone_passphrase = "clone_luks1_passphrase.bin"
+        rbd.create_passphrase_file(clone_passphrase)
+        if rbd.encrypt(clone_spec, encryption_type, clone_passphrase):
+            log.error(f"Apply RBD encryption {encryption_type} failed on {image_spec}")
+
+        encryption_config.append({encryption_type: clone_passphrase})
+        out, err = rbd.device_map(clone_spec, "nbd", reversed(encryption_config))
+        log.info(f"Out, Err : {out}: {err}")
+        if err:
+            log.error(
+                f"RBD device map failed for {image_spec} and encryption_config: {encryption_config}"
+            )
+
+        kw["config"]["device_names"].append(out)
+        kw["config"]["image_spec"].append(clone_spec)
+        krbd_io_handler(**kw)
+
+        flatten_config = {"encryption_config": reversed(encryption_config)}
+        if rbd.flatten_clone(pool, clone, **flatten_config):
+            log.error(f"Flatten clone with encryption failed for {clone_spec}")
+
+        log.info(
+            f"Verifying data integrity between original image: {image_spec} and clone: {clone_spec}"
+        )
+        check_data(rbd, image_spec, clone_spec)
+
+    except RbdBaseException as error:
+        print(error.message)
+        return 1
+
+
+def cleanup(rbd_obj, **kw):
```

Comment:
can we move this method to `rbd_utils` as it's more common functionality others can use it?

Reply:
@sunilangadi2 @manasagowri  we already have cleanup() method in rbd_utils, can we make use of that.

Reply:
@rkhambad The existing clean_up method expects all the pools to be cleaned up to be passed as inputs to it. Infact inside the new cleanup method which I have written here I have used the same existing cleanup() method in rbd_utils. 

The new method was written because I wanted to cleanup all the pools created with initial_rbd_config

@sunilangadi2 Yes the method looks generic and may be moved into rbd_utils. I will try doing that if we can first agree that two methods are required for this.

Reply:
ok sure if this method required as encapsulated to other utils method then let's add it in `rbd_utils`